check_whitespace.c

```diff
@@ -66,7 +67,7 @@ int is_clean(char* str) {
   // 0 if they're equal, and a positive value if the first is
   // greater than the second.
   result = strcmp(str, cleaned);
-
+  free(cleaned);  
```

Comment:
Valgrind gives you an error related to this line:

```
==6944== Invalid free() / delete / delete[] / realloc()
==6944==    at 0x100110A0D: free (in /usr/local/Cellar/valgrind/HEAD-6049595/lib/valgrind/vgpreload_memcheck-amd64-darwin.so)
==6944==    by 0x100000DC1: is_clean (check_whitespace.c:70)
==6944==    by 0x100000E3F: main (check_whitespace.c:88)
==6944==  Address 0x100000f36 is in the Text segment of ./check_whitespace
==6944==    at 0x100000F36: main (in ./check_whitespace)
```

The problem is that up on line 36 where, under certain circumstances, we the line `return ""` returns a _statically_ allocated (empty) string. A line like `return ""` doesn't used `malloc`/`calloc` to allocate that string, so the system allocates "statically", which means you can't `free()` it.

So you need to wrap that in an `if`:

```suggestion
  if (strcmp(cleaned, "") != 0) {
    free(cleaned);
  }
```